connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java

```diff
@@ -98,6 +118,12 @@ public class ConnectorConfig extends AbstractConfig {
             new InstantiableClassValidator()
```

Comment:
The KIP includes this detail:

> * If the .version property is left empty, or if there are no installed plugins with a matching name, errors will only be attributed to the plugin class property, not to the .version  property.
> * If a plugin class name is valid (at least one version is installed) but the non-empty .version property doesn't include any of the installed versions, then the error will be attributed to both the .version  property and the plugin class property.

I don't think the validators for the class or version configs are doing this right now. Is that part of this PR or a later PR?

Reply:
I have addressed this for converters and transformations (PR #3), but it's not that straightforward for connector and connector versions. The return on validation is a set of config Infos which is a combination of some predefined ones and ones defined in the connector. If the version is incorrect then the classloading of the connector itself will fail, so it is unclear as to which set of config infos we show. If we validate and show the config infos from the latest version of the connector, with the error attributed to the version config, the set of configs returned may not be accurate for the version requested. 

One possible way would be to return the common `ConnectorConfig` config def, but that is not something the API does atm, for instance we could do the same thing for invalid connector class, but we just throw an exception right now, which is why I have kept it the same here.

Reply:
> I have addressed this for converters and transformations (PR # 3)

Do you mean transformations and predicates? I see that there's some validation happening there, but I didn't see anything similar for converters in this PR.

> but it's not that straightforward for connector and connector versions

The KIP includes an exception for connector.class; we don't have to provide validation or recommendations for that.

We provide validation results for connector.plugin.version, because that's going to be the single most useful configuration in this new feature.

Reply:
I will see how we can return the validation result with the common connector configuration. This will have the config defs provided in `ConnectorConfig` but will not have the additional ones in the subclasses or the actual connector.